### PR TITLE
LPS-118531 - Create setSearchParamsWithoutPageReload function to ensure that page won't reload after URL changes

### PR DESCRIPTION
**Pull request by brunofariasdeo**
https://issues.liferay.com/browse/LPS-118531

### CODE REVIEW

**Comment by brunofariasdeo** [comment]
ci:test:sf

**Comment by brunofariasdeo** [comment]
ci:test:relevant

**Comment by brunofariasdeo** [comment]
ci:test:sf

**Comment by brunofariasdeo** [comment]
ci:test:relevant

**Comment by brunofariasdeo** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js [hunk 1]
```diff
@@ -29,21 +29,31 @@ export default ({fields}) => {
 		}
 	});
 
+	const setSearchParamsWithoutPageReload = (param, index) => {
+		const url = new URL(location.toString());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js:33]: @brunofariasdeo, please add a blank line after the variables declarations and the use of them. 😉

In addition, can you try to solve this issue without use the `URL` object? Unfortunately, it's not supported by IE, as you can see on https://developer.mozilla.org/en-US/docs/Web/API/URL/URL.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js [hunk 2]
```diff
@@ -29,21 +29,31 @@ export default ({fields}) => {
...
+		url.searchParams.set(param, index);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js:34]: @brunofariasdeo, please add a blank line after this statement.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js [hunk 3]
```diff
@@ -29,21 +29,31 @@ export default ({fields}) => {
...
+		window.history.replaceState({path: url.toString()}, '', url.toString());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js:35]: @brunofariasdeo, can we add blank spaces after `{` and before `}`? Did the SF check complain about it? In addition, do we need to replace the state of browser's history to make it work?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js [hunk 4]
```diff
@@ -29,21 +29,31 @@ export default ({fields}) => {
...
-				>
+				<li key={`card-item-${index}`}>
 					<a
 						className={`${
 							itemSelectedIndex == index ? 'selected' : ''
 						}`}
 						data-senna-off
 						href={`#${portletNamespace}card_${index}`}
+						onClick={() => {
+							setItemSelectedIndex(index);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/card-shortcut/CardShortcut.es.js:51]: @brunofariasdeo, please add a blank line after this statement.

**Pull request closed without merging**